pkg/pxc/queries/queries.go

```diff
@@ -315,8 +311,17 @@ func (p *Database) PresentInHostgroups(host string) (bool, error) {
 	return true, nil
 }
 
-func (p *Database) PrimaryHost() (string, error) {
+func (p *Database) PrimaryHost(scheduler string) (string, error) {
 	var host string
+	var writerID int
+	if scheduler == "percona" {
+		writerID = 100
+	} else {
+		// value of writer group is hardcoded in ProxySQL config inside docker image
+		// https://github.com/percona/percona-docker/blob/pxc-operator-1.3.0/proxysql/dockerdir/etc/proxysql-admin.cnf#L23
+		writerID = 11
+	}
```

Comment:
You can loose this else block: 

```
	// value of writer group is hardcoded in ProxySQL config inside docker image
	// https://github.com/percona/percona-docker/blob/pxc-operator-1.3.0/proxysql/dockerdir/etc/proxysql-admin.cnf#L23
	var writerID = 11
	if scheduler == "percona" {
		writerID = 100
	}
```

Also you can move `host` var just above the line where you first use ti.

Reply:
Thanks! Fixed.